sdk/src/main/java/com/qonversion/android/sdk/internal/di/misc/MiscAssemblyImpl.kt

```diff
@@ -0,0 +1,216 @@
+package com.qonversion.android.sdk.internal.di.misc
+
+import android.app.Application
+import android.content.Context
+import android.content.SharedPreferences
+import androidx.annotation.VisibleForTesting
+import com.qonversion.android.sdk.internal.InternalConfig
+import com.qonversion.android.sdk.internal.appState.AppLifecycleObserver
+import com.qonversion.android.sdk.internal.appState.AppLifecycleObserverImpl
+import com.qonversion.android.sdk.internal.common.BASE_API_URL
+import com.qonversion.android.sdk.internal.common.PREFS_NAME
+import com.qonversion.android.sdk.internal.common.localStorage.LocalStorage
+import com.qonversion.android.sdk.internal.common.localStorage.SharedPreferencesStorage
+import com.qonversion.android.sdk.internal.common.mappers.EntitlementMapper
+import com.qonversion.android.sdk.internal.common.mappers.ProductMapper
+import com.qonversion.android.sdk.internal.common.mappers.SubscriptionMapper
+import com.qonversion.android.sdk.internal.common.mappers.UserMapper
+import com.qonversion.android.sdk.internal.common.mappers.UserPropertiesMapper
+import com.qonversion.android.sdk.internal.common.mappers.UserPurchaseMapper
+import com.qonversion.android.sdk.internal.common.mappers.error.ApiErrorMapper
+import com.qonversion.android.sdk.internal.common.mappers.error.ErrorResponseMapper
+import com.qonversion.android.sdk.internal.common.serializers.JsonSerializer
+import com.qonversion.android.sdk.internal.common.serializers.Serializer
+import com.qonversion.android.sdk.internal.logger.ConsoleLogger
+import com.qonversion.android.sdk.internal.logger.Logger
+import com.qonversion.android.sdk.internal.networkLayer.RetryPolicy
+import com.qonversion.android.sdk.internal.networkLayer.apiInteractor.ApiInteractor
+import com.qonversion.android.sdk.internal.networkLayer.apiInteractor.ApiInteractorImpl
+import com.qonversion.android.sdk.internal.networkLayer.headerBuilder.HeaderBuilder
+import com.qonversion.android.sdk.internal.networkLayer.headerBuilder.HeaderBuilderImpl
+import com.qonversion.android.sdk.internal.networkLayer.networkClient.NetworkClient
+import com.qonversion.android.sdk.internal.networkLayer.networkClient.NetworkClientImpl
+import com.qonversion.android.sdk.internal.networkLayer.requestConfigurator.RequestConfigurator
+import com.qonversion.android.sdk.internal.networkLayer.requestConfigurator.RequestConfiguratorImpl
+import com.qonversion.android.sdk.internal.networkLayer.retryDelayCalculator.ExponentialDelayCalculator
+import com.qonversion.android.sdk.internal.networkLayer.retryDelayCalculator.RetryDelayCalculator
+import java.util.Locale
+import kotlin.random.Random
+
+internal object MiscAssemblyImpl : MiscAssembly {
+
+    override lateinit var application: Application
+
+    override val internalConfig: InternalConfig
+        get() = provideInternalConfig()
```

Comment:
Do we need a separate method to call from `get`? I think it would be more readable if we place the body right here, wouldn't it?